src/mjolnir/pbfgraphparser.cc

```diff
@@ -865,13 +865,14 @@ struct graph_callback : public OSMPBF::Callback {
       results = results ? results : empty_node_results_;
     }
 
-    const auto& highway_junction = results->find("highway");
```

Comment:
this looks like a lot of diffs but its really the same code just simplified. there were:

1. tons of copy paste of checking the length of the value, i made a single varaible for it and did it once at the top of the loop
2. an insane amount of ternaries around things there were already true/false values (wtf?)
3. lots of nested `ifs` where were very simply combined into a single `if` with some `&&`s
4. iterators declared as const reference (wtf?)
5. main change was that we store a bool for when a junction could be named and then at the end, only once outside of the loop, if the junction actually had a name we set the flag that eventually gets written into nodeinfo (removing false positives as a possibility)